lib/sshkit/backends/abstract.rb

```diff
@@ -122,7 +122,8 @@ def create_command_and_execute(args, options)
       end
 
       def command(args, options)
-        SSHKit::Command.new(*[*args, options.merge({in: @pwd.nil? ? nil : File.join(@pwd), env: @env, host: @host, user: @user, group: @group})])
+        pwd = (!@pwd || @pwd.empty?) ? nil : File.join(@pwd)
```

Comment:
Can you factor this out into a private helper method, e.g. `pwd_path`? Also, `@pwd.nil?` is more readable than `!@pwd` IMHO.